### PR TITLE
template context additions

### PR DESCRIPTION
**Pull request by nbuonin**
- Implement Course description field, custom template tags
- Implement Activity status annotation

### CODE REVIEW

File: locustempus/templates/main/course_detail.html [hunk 1]
```diff
@@ -64,6 +65,10 @@
 {% endif %}
 
 <div class="row dashboard-container" data-cy="project-list">
+    <p>Authors: {{ course | course_faculty_string }}</p>
```
Comment (sdreher) [inline locustempus/templates/main/course_detail.html:68]: This looks quite wonky from a UI perspective, and disrupts the card rendering in the flex box. I'm hesitant to merge this as is...maybe hide this information for the moment?

File: locustempus/templates/main/course_edit.html [hunk 2]
```diff
@@ -14,8 +16,17 @@
...
+            <label for="{{ form.description.id_for_label }}">Description:</label>
+            {% if form.description.errors %}
+                {{ form.description|add_class:"form-control is-invalid"|attr:"data-cy:workspace-description" }}
+                <div class="invalid-feedback">
+                    {% for error in form.description.errors %}
+                        {{ error }}
+                    {% endfor %}
+                </div>
+            {% else %}
+                <textarea name="id_description" cols="40" rows="3" class="form-control" maxlength="200" placeholder="A brief summary of your workspace." id="id_description" data-cy="field-workspace-description">{{ object | course_description }}</textarea>
```
Comment (sdreher) [inline locustempus/templates/main/course_edit.html:28]: The course description does not appear to be saving. To reproduce: Go to Edit Course, change the Description, hit save. (Navigate back to the workspace detail page.) Go to Edit Course -- the description was not saved.

File: locustempus/templates/main/course_create.html [hunk 3]
```diff
@@ -29,8 +29,17 @@
...
+            <label for="{{ form.description.id_for_label }}">Description:</label>
+            {% if form.description.errors %}
+                {{ form.description|add_class:"form-control is-invalid"|attr:"data-cy:workspace-description" }}
+                <div class="invalid-feedback">
+                    {% for error in form.description.errors %}
+                        {{ error }}
+                    {% endfor %}
+                </div>
+            {% else %}
+                <textarea name="id_description" cols="40" rows="3" class="form-control" maxlength="200" placeholder="A brief summary of your workspace." id="id_description" data-cy="field-workspace-description"></textarea>
```
Comment (sdreher) [inline locustempus/templates/main/course_create.html:41]: The name field here should be `description` ... this is why the CourseForm is not picking up the description.

File: locustempus/templates/main/course_edit.html (hunk 2 above)
Comment (sdreher) [reply locustempus/templates/main/course_edit.html:28]: Note: See the comment above about the `id_description`. Fix that name field should fix this.

File: locustempus/templates/main/course_edit.html [hunk 4]
```diff
@@ -14,8 +16,17 @@
...
+            <label for="{{ form.description.id_for_label }}">Description:</label>
+            {% if form.description.errors %}
+                {{ form.description|add_class:"form-control is-invalid"|attr:"data-cy:workspace-description" }}
+                <div class="invalid-feedback">
+                    {% for error in form.description.errors %}
+                        {{ error }}
+                    {% endfor %}
+                </div>
+            {% else %}
+                <textarea name="description" cols="40" rows="3" class="form-control" maxlength="200" placeholder="A brief summary of your workspace." id="{{ form.description.id_for_label }}" data-cy="field-workspace-description"></textarea>
```
Comment (sdreher) [inline locustempus/templates/main/course_edit.html:28]: The textarea should be initialized with a description. I'm going to PR this change.

Comment (nbuonin) [reply locustempus/templates/main/course_edit.html]: Thank you, I deleted it when I updated the field.

File: locustempus/main/forms.py [hunk 5]
```diff
@@ -1,9 +1,32 @@
...
 from django.core.validators import RegexValidator
 
 
+class CourseForm(forms.ModelForm):
+    description = forms.CharField(widget=forms.Textarea, required=False)
+
+    def save(self, commit=True):
+        course = super().save(commit=False)
+        description = self.cleaned_data.get('description', None)
+        if description:
```
Comment (sdreher) [inline locustempus/main/forms.py:14]: This test disallows the ability to clear out the description. A better test would be `if description is not None`. I'll make the change.

**Pull request merged**